### PR TITLE
Update to openj9-0.9 and Extensions version 9.0.0.1

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Signed-off-by: andrew-m-leonard <andrew_m_leonard@uk.ibm.com>

### CODE REVIEW

**Comment by andrew-m-leonard** [comment]
```
admin@leo01:~/mydocker/ccm/jdk9Public$ build/linux-x86_64-normal-server-release/images/jdk/bin/java -version
openjdk version "9.0.4"
OpenJDK Runtime Environment (build 9.0.4+0)
Eclipse OpenJ9 VM (build openj9-0.9.0-rc1, JRE 9 Linux amd64-64-Bit Compressed References 20180619_000000 (JIT enabled, AOT enabled)
OpenJ9   - da56ab5e
OMR      - da0c7eeb
JCL      - f8f07f1d91 based on jdk-9.0.4+12)
```
```
admin@leo01:~/mydocker/ccm/jdk9Public$ build/linux-x86_64-normal-server-release/images/jdk/bin/java -Xinternalversion
Eclipse OpenJ9 OpenJDK 64-bit Server VM (9.0.4+0) from linux-amd64 JRE with Extensions for OpenJDK for Eclipse OpenJ9 9.0.0.1, built on Jun 19 2018 15:22:43 by admin with g++ (Ubuntu 4.8.4-2ubuntu1~14.04.4) 4.8.4
```

File: closed/autoconf/custom-spec.gmk.in [hunk 1]
```diff
@@ -34,6 +34,9 @@ endef
 OPENJ9_TOPDIR           := @OPENJ9_TOPDIR@
 OPENJ9OMR_TOPDIR        := @OPENJ9OMR_TOPDIR@
 
+J9JDK_EXT_VERSION       := 9.0.0.1
```
Comment (pshipton) [inline closed/autoconf/custom-spec.gmk.in:37]: Irina used 9.0.0.4 as the release version, not sure why.

Comment (pshipton) [reply closed/autoconf/custom-spec.gmk.in:37]: @irinarada ^

File: closed/get_j9_source.sh [hunk 2]
```diff
@@ -51,10 +51,10 @@ declare -A git_urls
 declare -A shas
 
 git_urls[openj9]=https://github.com/eclipse/openj9
-branches[openj9]=master
+branches[openj9]=openj9-0.9.0-rc1
```
Comment (pshipton) [inline closed/get_j9_source.sh:54]: Shouldn't these changes be made after branching?

File: closed/autoconf/custom-spec.gmk.in (hunk 1 above)
Comment (andrew-m-leonard) [reply closed/autoconf/custom-spec.gmk.in:37]: oh!

File: closed/get_j9_source.sh (hunk 2 above)
Comment (andrew-m-leonard) [reply closed/get_j9_source.sh:54]: wasn't sure we were going to bother branching jdk9, since this is the last ever release....

Comment (pshipton) [reply closed/get_j9_source.sh:54]: Interesting idea. @DanHeidinga @mstoodle any thoughts? It does make it clear that no further changes need to be made to the jdk9 extensions, even when OpenJ9 changes, such as for https://github.com/eclipse/openj9/pull/2189

Comment (DanHeidinga) [reply closed/get_j9_source.sh:54]: My preference is to have a `openj9-0.9.0` branch here so it's very clear what content was in the 0.9.0 release.  This makes it possible to rebuild the release if any changes are delivered to this repo after the 0.9 release.

It's also a consistency thing: all the extensions repos included in a release should have a branch for that release.  Naming the release branches consistently saves anyone from having to think about which branch to use.

Comment (mstoodle) [reply closed/get_j9_source.sh:54]: The clarity reasoning is the most compelling. For relatively little work, anyone looking back on this repo will be able to understand why this change was made if it's in a branch specific to 0.9.0 .

Comment (andrew-m-leonard) [reply closed/get_j9_source.sh:54]: Agree, i'll do the branch today

**Comment by pshipton** [comment]
Closing as the decision was not to change the OpenJ9 branch from master

**Pull request closed without merging**